algo/src/main/java/org/neo4j/graphalgo/LabelPropagationProc.java

```diff
@@ -49,22 +50,17 @@
             @Name(value = "direction", defaultValue = "OUTGOING") String directionName,
             @Name(value = "config", defaultValue = "{}") Map<String, Object> config) {
 
+
+        ProcedureConfiguration configuration = ProcedureConfiguration.create(config);
+
         label = emptyToNull(label);
         relationshipType = emptyToNull(relationshipType);
         Direction direction = parseDirection(directionName);
 
-        int iterations = ((Number) config.getOrDefault(
-                CONFIG_ITERATIONS,
-                DEFAULT_ITERATIONS)).intValue();
-        String partitionProperty = emptyToNull((String) config.getOrDefault(
-                CONFIG_PARTITION_KEY,
-                DEFAULT_PARTITION_KEY));
-        String weightProperty = emptyToNull((String) config.getOrDefault(
-                CONFIG_WEIGHT_KEY,
-                DEFAULT_WEIGHT_KEY));
-        boolean write = (boolean) config.getOrDefault(
-                CONFIG_WRITE,
-                DEFAULT_WRITE);
+        final int iterations = configuration.getIterations(DEFAULT_ITERATIONS);
```

Comment:
We should put defaults into the configuration class itself (and make them overridable *if needed*)